lib/client.js

```diff
@@ -1331,6 +1331,65 @@ MatrixClient.prototype.invite = function(roomId, userId, callback) {
 };
 
 /**
+ * Invite a user to a room based on their email address.
+ * @param {string} roomId The room to invite the user to.
+ * @param {string} email The email address to invite.
+ * @param {string} displayName The display name to use. If omitted, everything
+ * before the '@' in the email address is used.
+ * @param {module:client.callback} callback Optional.
+ * @return {module:client.Promise} Resolves: TODO
+ * @return {module:http-api.MatrixError} Rejects: with an error response.
+ */
+MatrixClient.prototype.inviteByEmail = function(roomId, email, displayName, callback) {
+    if (utils.isFunction(displayName)) {
+        callback = displayName; displayName = undefined;
+    }
+    if (!displayName) {
+        displayName = email.split("@")[0];
```

Comment:
> Maybe we should just use the email address verbatim for the diaplayname here?

We specifically do not want to use the email address for this because it will leak the email to user ID mapping as per http://matrix.org/speculator/spec/head#id43

> if we do split it, we should probably catch it not containing an '@'.

What would you prefer here? If `@` is not present, this will default to the entire string:

```javascript
> "foo".split("@")[0]
'foo'
```